stl/inc/xutility

```diff
@@ -376,17 +375,16 @@ struct _Ref_fn { // pass function object by value as a reference
 };
 
 template <class _Fn>
-_INLINE_VAR constexpr bool _Pass_functor_by_value_v = conjunction_v<bool_constant<sizeof(_Fn) <= sizeof(void*)>,
+_INLINE_VAR constexpr bool _Pass_functor_by_value = conjunction_v<bool_constant<sizeof(_Fn) <= sizeof(void*)>,
     is_trivially_copy_constructible<_Fn>, is_trivially_destructible<_Fn>>;
 
-template <class _Fn, enable_if_t<_Pass_functor_by_value_v<_Fn>, int> = 0> // TRANSITION, if constexpr
-constexpr _Fn _Pass_fn(_Fn _Val) { // pass functor by value
-    return _Val;
-}
-
-template <class _Fn, enable_if_t<!_Pass_functor_by_value_v<_Fn>, int> = 0>
-constexpr _Ref_fn<_Fn> _Pass_fn(_Fn& _Val) { // pass functor by "reference"
-    return {_Val};
+template <class _Fn>
+constexpr auto _Pass_fn(_Fn& _Func) {

```

Comment:
Can we strengthen the noexcept specification by using `is_nothrow_copy_constructible_v || _Pass_functor_by_value`? 

And yes both suggestions seem exclusive

Reply:
No, but we can make it unconditionally `noexcept`. If `_Pass_functor_by_value` is true, we trivially-copy-construct the return value. If it's not true, we aggregate-init a `_Ref_fn` with a reference member. Neither case throws exceptions.